File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Repository/TratamientoRepository.java
package com.example.BackendCitasMedicas.Repository;

import com.example.BackendCitasMedicas.Model.Tratamiento;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TratamientoRepository extends JpaRepository<Tratamiento, Long> {
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Repository/TratamientoRepository.kt
package com.example.frontcitasmedicas.Repository

import com.example.frontcitasmedicas.Model.Tratamiento
import com.example.frontcitasmedicas.Service.RetrofitClient

class TratamientoRepository {

    private val api = RetrofitClient.tratamientoService

    suspend fun getTratamientos() = api.getAllTratamientos()

    suspend fun getTratamiento(id: Long) = api.getTratamientoById(id)

    suspend fun addTratamiento(tratamiento: Tratamiento) = api.addTratamiento(tratamiento)

    suspend fun updateTratamiento(tratamiento: Tratamiento) = api.updateTratamiento(tratamiento)

    suspend fun deleteTratamiento(id: Long) = api.deleteTratamiento(id)
}
